test/conftest.py

```
@@ -37,3 +43,111 @@ def fixture_video_embeddings_path(video_path: Path) -> Path:
+    ]
+    process = subprocess.Popen(go2rtc_command)
+    try:
+
+        def is_rtsp_stream_ready(rtsp_address):
+            """
+            Checks if the RTSP stream is ready by attempting to read from it.
+            """
+            try:
+                # Use ffmpeg to probe the stream.  A quick probe is usually sufficient.
+                ffmpeg.probe(rtsp_address, timeout=1)
+                return True
```

Ruff (0.11.9)

Consider moving this statement to an `else` block

(TRY300)